filters/auth/oidc_introspection_test.go

```diff
@@ -0,0 +1,295 @@
+			},
+			path:      "/path/asdf",
+			expected:  200,
+			expectErr: false,
+		},
+		{
+			msg: "path / permitted for group Purchasing-Department",
+			args: []interface{}{
+				"/:groups.#(%\"Purchasing-Department\")",
+			},
+			path:      "/",
+			expected:  200,
+			expectErr: false,
+		},
+		{
+			msg: "path /some/path permitted",
+			args: []interface{}{
+				"/some/path:groups.#(%\"Purchasing-Department\")",
+			},
+			path:      "/some/path/down/there",
+			expected:  200,
+			expectErr: false,
+		},
+		{
+			msg: "path /some/otherpath denied",
+			args: []interface{}{
+				"/some/path:groups.#(%\"Purchasing-Department\")",
+			},
+			path:      "/some/otherpath",
+			expected:  401,
+			expectErr: false,
+		},
+		{
+			msg: "wrong group denied",
+			args: []interface{}{
+				"/some/path:groups.#(%\"Shipping-Department\")",
+			},
+			path:      "/some/path",
+			expected:  401,
+			expectErr: false,
+		},
+		{
+			msg: "several queries, path matching",
+			args: []interface{}{
+				"/some/path:[@this].#(sub==\"somesub\")",
+				"/another/path:groups.#(%\"CD-*\")",
+				"/asdf/:groups.#(%\"CD-*\") 'groups.#(%\"*-Test-Users\")' groups.#(%\"Purchasing-Department\")",
+				"/:groups.#(%\"Purchasing-Department\")",
+			},
+			path:      "/asdf/asdf",
+			expected:  200,
+			expectErr: false,
+		},
+		{
+			msg: "several queries, no path matching",
+			args: []interface{}{
+				"/some/path:[@this].#(sub==\"somesub\")",
+				"/another/path:groups.#(%\"CD-*\")",
+				"/asdf/:groups.#(%\"CD-*\") 'groups.#(%\"*-Test-Users\")' groups.#(%\"Purchasing-Department\")",
+				"/abc:groups.#(%\"Purchasing-Department\")",
+			},
+			path:      "/xyz",
+			expected:  401,
+			expectErr: false,
+		},
+	} {
+		t.Run(tc.msg, func(t *testing.T) {
+			backend := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				t.Logf("backend got request: %+v", r)
+				w.Write([]byte("OK"))
+			}))
+			defer backend.Close()
+			t.Logf("backend URL: %s", backend.URL)
+
+			spec := &tokenOidcSpec{
+				typ:             checkOIDCAnyClaims,
+				SecretsFile:     "/tmp/foo",
+				secretsRegistry: secrettest.NewTestRegistry(),
+			}
+			fr := make(filters.Registry)
+			fr.Register(spec)
+			dc := testdataclient.New(nil)
+			proxy := proxytest.WithRoutingOptions(fr, routing.Options{
+				DataClients: []routing.DataClient{dc},
+				Log:         loggingtest.New(),
+			})
+			defer proxy.Close()
+			reqURL, err := url.Parse(proxy.URL)
+			if err != nil {
+				t.Errorf("Failed to parse url %s: %v", proxy.URL, err)
+			}
+			reqURL.Path = tc.path
+			oidcServer := createOIDCServer(proxy.URL+"/redirect", validClient, "mysec")
+			defer oidcServer.Close()
+			t.Logf("oidc/auth server URL: %s", oidcServer.URL)
+			// create filter
+			sargs := []interface{}{
+				validClient,
+				"mysec",
+				proxy.URL + "/redirect",
+			}
+			sargs = append([]interface{}{oidcServer.URL}, sargs...)
+			sargs = append(sargs, testKey, testKey)
```

Comment:
Any reason why not:
```
sargs := []interface{}{
	oidcServer.URL,  
	validClient,
	"mysec",
	proxy.URL + "/redirect",
	testKey,
	testKey,
}
```

Reply:
That's a very valid point - guess copy-pasta 😄 thanks